src/app.ts

```diff
@@ -14,195 +14,423 @@
+        AlgorandApp._params.cla,
+        AlgorandApp._INS.GET_PUBLIC_KEY,
+        p1,
+        0,
+        data
+      )
+
+      const response = processResponse(responseBuffer)
+
+      const pubkey = response.readBytes(PUBKEYLEN)
+      const address = response.getAvailableBuffer().toString()
+
+      return {
+        publicKey: Buffer.from(pubkey),
+        address: Buffer.from(address),
+        returnCode: LedgerError.NoErrors,
+        errorMessage: ERROR_DESCRIPTION[LedgerError.NoErrors],
+        // Legacy
+        bech32_address: Buffer.from(address),
+        compressed_pk: Buffer.from(pubkey),
+        return_code: LedgerError.NoErrors,
+        error_message: ERROR_DESCRIPTION[LedgerError.NoErrors],
+      } as ResponseAddress
+    } catch (e) {
+      throw processErrorResponse(e)
+    }
+  }
+
+  async signData( signingData: StdSigData,
+    metadata: StdSignMetadata
+  ): Promise<StdSigDataResponse> {
+    let dataToEncode
+    let decodedData
+
+    if (metadata.encoding === 'base64') {
+      decodedData = Buffer.from(signingData.data, 'base64')
+    } else {
+      throw new Error('Failed decoding')
     }
 
-    async sign(accountId = 0, message: string | Buffer): Promise<ResponseSign> {
-        const chunks = AlgorandApp.prepareChunksFromAccountId(accountId, message);
-
-        let p2 = (chunks.length > 1) ? AlgorandApp._params.p2Values.P2_MORE_CHUNKS : AlgorandApp._params.p2Values.P2_LAST_CHUNK;
-
-        try {
-            let signatureResponse = await this.sendGenericChunk(AlgorandApp._INS.SIGN_MSGPACK, p2, 0, chunks.length, chunks[0])
-
-            for (let i = 1; i < chunks.length; i += 1) {
-                p2 = (i < chunks.length - 1) ? AlgorandApp._params.p2Values.P2_MORE_CHUNKS : AlgorandApp._params.p2Values.P2_LAST_CHUNK;
-                signatureResponse = await this.sendGenericChunk(AlgorandApp._INS.SIGN_MSGPACK, p2, i, chunks.length, chunks[i])
-            }
-
-            return {
-                signature: signatureResponse.readBytes(signatureResponse.length()),
-            } as ResponseSign
+    const signerBuffer = Buffer.from(signingData.signer)
+    const scopeBuffer = Buffer.from([metadata.scope])
+    const encodingBuffer = serializeEncoding(metadata.encoding)
+    const dataBuffer = decodedData
+    let domainBuffer = Buffer.from([])
+    if(signingData.domain) {
+      domainBuffer = Buffer.from(signingData.domain)
+    }
+    let requestIdBuffer = Buffer.from([])
+    if (signingData.requestId) {
+      const requestIdHexStr = Buffer.from(signingData.requestId, 'base64').toString('hex');
+      requestIdBuffer = Buffer.from(requestIdHexStr, 'hex');
+    }
+    let authDataBuffer = Buffer.from([])
+    if (signingData.authenticationData) {
+      authDataBuffer = Buffer.from(signingData.authenticationData)
+    }
+    const pathBuffer = signingData.hdPath
+      ? this.serializePath(signingData.hdPath)
+      : this.serializePath("m/44'/283'/0'/0/0")
```

Comment:
Can we convert this to a const outside the function? Like "DEFAULT_SIGN_DATA_PATH"